### ISSUE TITLE
Dissapearing Planet Glitch

### ISSUE THREAD

**Issue by CaptainStalfoss** [zmaster587/AdvancedRocketry#1281]
## Version of Advanced Rocketry
1.12.2-1.4.0-80
## Version of LibVulpes
1.12.2-0.2.8-30
## Version of Minecraft
1.12.2
## Crash report or log (if applicable)
`<galaxy>
    <star name="Sol" temp="100" x="0" y="0" numPlanets="0" numGasGiants="0">
	<planet name="Mercury" DIMID="101" customIcon="MarsLike">
			<isKnown>true</isKnown>
		    	<biomeIds>advancedrocketry:hotdryrock,advancedrocketry:moon,advancedrocketry:moondark</biomeIds>
            		<skyColor>0,0,0</skyColor>
			<atmosphereDensity>20</atmosphereDensity>
			<gravitationalMultiplier>38</gravitationalMultiplier>
			<orbitalDistance>12</orbitalDistance>
			<rotationalPeriod>1400000</rotationalPeriod>
			<fillerBlock>biomesoplenty:dried_sand</fillerBlock>
			<oreGen>
				<ore block="advancedrocketry:hotturf" minHeight="1" maxHeight="160" clumpSize="20" chancePerChunk="10" />
				<ore block="minecraft:gravel" minHeight="1" maxHeight="68" clumpSize="33" chancePerChunk="8" />
				<ore block="moreplanets:fronos_block" meta="0" minHeight="1" maxHeight="160" clumpSize="12" chancePerChunk="20" />
				<ore block="moreplanets:fronos_block" meta="1" minHeight="1" maxHeight="160" clumpSize="12" chancePerChunk="20" />
				<ore block="advancedrocketry:geode" minHeight="1" maxHeight="160" clumpSize="12" chancePerChunk="15" />
				<ore block="mocreatures:mocdirt" meta="0" minHeight="30" maxHeight="160" clumpSize="12" chancePerChunk="20" />
				<ore block="minecraft:stone" meta="1" minHeight="1" maxHeight="68" clumpSize="33" chancePerChunk="10" />
				<ore block="minecraft:stone" meta="3" minHeight="1" maxHeight="68" clumpSize="33" chancePerChunk="10" />
				<ore block="minecraft:stone" meta="5" minHeight="1" maxHeight="68" clumpSize="33" chancePerChunk="10" />
				<ore block="erebus:volcanic_rock" minHeight="1" maxHeight="30" clumpSize="12" chancePerChunk="15" />
				<ore block="galacticraftplanets:venus" meta="2" minHeight="0" maxHeight="24" clumpSize="12" chancePerChunk="15" />
				<ore block="galacticraftplanets:venus" meta="3" minHeight="0" maxHeight="20" clumpSize="12" chancePerChunk="15" />
				<ore block="morefuelsmod:ore_lava" minHeight="1" maxHeight="11" clumpSize="8" chancePerChunk="15" />
				<ore block="moreplanets:fronos_ore" meta="0" minHeight="0" maxHeight="68" clumpSize="9" chancePerChunk="20" />
				<ore block="moreplanets:fronos_ore" meta="1" minHeight="0" maxHeight="32" clumpSize="10" chancePerChunk="1" />
				<ore block="moreplanets:fronos_ore" meta="2" minHeight="25" maxHeight="60" clumpSize="6" chancePerChunk="8" />
				<ore block="moreplanets:fronos_ore" meta="3" minHeight="0" maxHeight="68" clumpSize="6" chancePerChunk="10" />
				<ore block="moreplanets:fronos_ore" meta="4" minHeight="5" maxHeight="60" clumpSize="16" chancePerChunk="1" />
				<ore block="moreplanets:fronos_ore" meta="5" minHeight="1" maxHeight="65" clumpSize="5" chancePerChunk="8" />
				<ore block="moreplanets:fronos_ore" meta="7" minHeight="0" maxHeight="34" clumpSize="7" chancePerChunk="1" />
				<ore block="moreplanets:fronos_ore" meta="8" minHeight="0" maxHeight="16" clumpSize="8" chancePerChunk="1" />
				<ore block="moreplanets:fronos_ore" meta="10" minHeight="0" maxHeight="16" clumpSize="8" chancePerChunk="8" />
				<ore block="moreplanets:fronos_ore" meta="11" minHeight="0" maxHeight="63" clumpSize="4" chancePerChunk="10" />
				<ore block="moreplanets:fronos_ore" meta="12" minHeight="5" maxHeight="63" clumpSize="15" chancePerChunk="4" />
				<ore block="moreplanets:fronos_ore" meta="13" minHeight="0" maxHeight="32" clumpSize="9" chancePerChunk="2" />
				<ore block="libvulpes:ore0" minHeight="1" maxHeight="16" clumpSize="16" chancePerChunk="3" />
				<ore block="libvulpes:ore0" meta="8" minHeight="5" maxHeight="68" clumpSize="6" chancePerChunk="6" />
				<ore block="libvulpes:ore0" meta="10" minHeight="0" maxHeight="14" clumpSize="16" chancePerChunk="1" />
				<ore block="erebus:ore_silver" minHeight="0" maxHeight="32" clumpSize="9" chancePerChunk="3" />
				<ore block="erebus:ore_gneiss" minHeight="20" maxHeight="68" clumpSize="9" chancePerChunk="7" />
				<ore block="moreplanets:fronos_ore" meta="9" minHeight="0" maxHeight="60" clumpSize="4" chancePerChunk="3" />
				<ore block="biomesoplenty:gem_ore" meta="1" minHeight="0" maxHeight="60" clumpSize="4" chancePerChunk="3" />
				<ore block="biomesoplenty:gem_ore" meta="2" minHeight="0" maxHeight="60" clumpSize="4" chancePerChunk="3" />
				<ore block="biomesoplenty:gem_ore" meta="3" minHeight="0" maxHeight="60" clumpSize="4" chancePerChunk="3" />
				<ore block="biomesoplenty:gem_ore" meta="4" minHeight="0" maxHeight="60" clumpSize="4" chancePerChunk="3" />
				<ore block="biomesoplenty:gem_ore" meta="5" minHeight="0" maxHeight="60" clumpSize="4" chancePerChunk="3" />
				<ore block="biomesoplenty:gem_ore" meta="6" minHeight="0" maxHeight="60" clumpSize="4" chancePerChunk="3" />
				<ore block="tropicraft:ore" meta="1" minHeight="0" maxHeight="60" clumpSize="4" chancePerChunk="3" />
				<ore block="tropicraft:ore" meta="2" minHeight="0" maxHeight="60" clumpSize="4" chancePerChunk="3" />
			</oreGen>
		        </planet>
		<planet name="Venus" DIMID="-31" dimMapping="" customIcon="GasGiantRed">
				<isKnown>true</isKnown>
				<orbitalDistance>32</orbitalDistance>
				<gravitationalMultiplier>100</gravitationalMultiplier>
				<atmosphereDensity>190</atmosphereDensity>
				</planet>
        	<planet name="Earth" DIMID="0" dimMapping="" customIcon="earthlike">
			<isKnown>true</isKnown>
			<fogColor>1.0,1.0,1.0</fogColor>
			<skyColor>1.0,1.0,1.0</skyColor>
			<gravitationalMultiplier>100</gravitationalMultiplier>
			<orbitalDistance>52</orbitalDistance>
			<orbitalPhi>0</orbitalPhi>
			<rotationalPeriod>24000</rotationalPeriod>
			<atmosphereDensity>100</atmosphereDensity>
			<planet name="Moon" DIMID="-28" dimMapping="" customIcon="moon">
				<isKnown>true</isKnown>
				<orbitalDistance>150</orbitalDistance>
				<gravitationalMultiplier>100</gravitationalMultiplier>
				<atmosphereDensity>0</atmosphereDensity>
				</planet>
			</planet>
		<planet name="Mars" DIMID="-29" dimMapping="" customIcon="MarsLike">
				<isKnown>true</isKnown>
				<orbitalDistance>70</orbitalDistance>
				<gravitationalMultiplier>100</gravitationalMultiplier>
				<atmosphereDensity>6</atmosphereDensity>
				</planet>
		<planet name="Asteroid Belt" DIMID="-30" dimMapping="" customIcon="IceWorld">
				<isKnown>true</isKnown>
				<orbitalDistance>80</orbitalDistance>
				<gravitationalMultiplier>100</gravitationalMultiplier>
				<atmosphereDensity>0</atmosphereDensity>
				</planet>
		<planet name="Stupiter" customIcon="GasGiantRed">
				<isKnown>true</isKnown>
            			<GasGiant>true</GasGiant>
            			<gas>nitrogen</gas>
            			<gas>oxygen</gas>
				<orbitalDistance>110</orbitalDistance>
				<rotationalPeriod>10600</rotationalPeriod>
				<gravitationalMultiplier>165</gravitationalMultiplier>
				<atmosphereDensity>200</atmosphereDensity>
				</planet>
		<planet name="Jupiter" customIcon="GasGiantRed">
				<isKnown>true</isKnown>
            			<GasGiant>true</GasGiant>
            			<gas>nitrogen</gas>
            			<gas>oxygen</gas>
				<orbitalDistance>120</orbitalDistance>
				<rotationalPeriod>10600</rotationalPeriod>
				<gravitationalMultiplier>165</gravitationalMultiplier>
				<atmosphereDensity>200</atmosphereDensity>
				</planet>
		<planet name="Saturn" customIcon="GasGiantRed">
				<isKnown>true</isKnown>
            			<GasGiant>true</GasGiant>
            			<gas>nitrogen</gas>
            			<gas>oxygen</gas>
            			<hasRings>true</hasRings>
				<orbitalDistance>140</orbitalDistance>
				<rotationalPeriod>10600</rotationalPeriod>
				<gravitationalMultiplier>150</gravitationalMultiplier>
				<atmosphereDensity>200</atmosphereDensity>
				</planet>
		<planet name="Uranus" customIcon="GasGiantBlue">
				<isKnown>true</isKnown>
            			<GasGiant>true</GasGiant>
            			<gas>nitrogen</gas>
            			<gas>oxygen</gas>
            			<hasRings>true</hasRings>
				<orbitalDistance>160</orbitalDistance>
				<rotationalPeriod>17250</rotationalPeriod>
				<gravitationalMultiplier>130</gravitationalMultiplier>
				<atmosphereDensity>200</atmosphereDensity>
				</planet>
		<planet name="Neptune" customIcon="GasGiantBlue">
				<isKnown>true</isKnown>
            			<GasGiant>true</GasGiant>
            			<gas>nitrogen</gas>
            			<gas>oxygen</gas>
				<orbitalDistance>180</orbitalDistance>
				<rotationalPeriod>21100</rotationalPeriod>
				<gravitationalMultiplier>130</gravitationalMultiplier>
				<atmosphereDensity>200</atmosphereDensity>
        </planet>
    </star>
</galaxy>`

## Description of the problem
hi, love the mod so far, it's amazing.
Anyway, I am working on creating the Solar system, its currently WIP, But For some reason, Jupiter keeps not showing up.
Or "Stupiter" as i call it in the log, because the only way to make it work was to keep the broken gas giant, and make a new Jupiter.

My only guess is that the first gas giant in the list breaks for some reason.
it doesn't matter how i putt the code or replaced it, the first gas giant will not show up.

(On a side note, 200 limit on planet distance is a bit small, you really have to squeeze all the planets in there.)

Short version:
Planet called "Stupiter" in the code will not appear.

**Comment by zmaster587** [comment]
if you run `/advRocketry planet list` does it come up on the list?

**Comment by CaptainStalfoss** [comment]
No, i tested it just now and it did not show up.

**Comment by zmaster587** [comment]
Can you upload the temp.dat?

**Comment by CaptainStalfoss** [comment]
This one?
https://www.dropbox.com/s/pjvhjk2ddud2p8a/temp.dat?dl=0

**Comment by zmaster587** [comment]
yeah, that's the one.   That's also kinda strange, I copied the XML into a new world for me and it's showing up fine.  I might expect it to possibly be an issue if it were an existing world (that was supposed to be fixed a few versions back, but edges cases sometimes pop up).  I'll take a look at the temp.dat.  

In the meantime can you test to see if this occurs with only libvulpes and AR installed?  Could you also post your startup log?

**Comment by zmaster587** [comment]
Update:  Tested the XML with the following mods installed:
Advaned rocketry 1.12.2-1.4.0-80
LibVulpes 1.12.2-1.4.0-80
Galacticraft-Planets-1.12.2-4.0.1.176
GalacticraftCore-1.12.2-4.0.1.176
MicdoodleCore-1.12.2-4.0.1.176

It's showing up for me as dimID2 as seen in the attached screenshot.
![image](https://user-images.githubusercontent.com/3536685/39669489-3fea47fc-50bb-11e8-8156-fb0e2fdaa35c.png)

I noticed that your temp.dat was lacking a dim 2, at this point I'm not certain why.  The only difference I can think of at this point is the possibility you have some mod that's using dimid 2 already, in which case you may either want to change the dimension that mod uses or change minDimension in advRocketry.cfg.

Another theory I have (with the info I've been given so far) is that possibly minDimension get set to 1.  finally, there may be file corruption of the advrocketry jar file (unlikely though).

**Comment by CaptainStalfoss** [comment]
Oh you know what it is, Dim 2 has some conflict i think with the "flans apocalypse mod"
its why i don't use Dim 2 in my pack, but i guess the gas giant auto chose it.
So solution would be to set gas giant dimensions manually.

overall, Dim 2 being broken is an issue with Flans mod, not this one.
